### PR TITLE
JG API updates

### PR DESCRIPTION
**Pull request by rowanhogan**
- Use link service for campaign donateUrl since Luis "fixed" it!
- Hack to include offline amounts in JG fundraising leaderboards 🤷

Screen Shot 2020-04-20 at 5 47 49 pm

### CODE REVIEW

**Comment by danhenderson** [comment]
When fetching and merging the offline amounts, does the `fetchPages` call make a request per page or just a single request. I'm obviously thinking from a performance and API load POV. I had a quick glance in `fetchPages`, and I guess the answer is it depends. And given that `fetchPages` behaves differently depending on whether you are fetching for an event, or a campaign, will the offline amount always be available regardless of what sort of leaderboard you are trying to get? Or are we just concerning ourselves with the main use case of a campaign?

My second point isn't an issue, just a cosmetic thing, so not a show stopper by any stretch. But I feel like when there is a big chunk of logic in the middle of a promise chain, it can become hard to follow. We could look at pulling that into it's own function so that the main flow of the promise chain just looks something like `.then(results => this.mergeOfflineAmounts(results))` or something like that. Then if you need to drill into that method, you can, but otherwise it just reads nicely. You know I love a good promise chain that is close to single lines all the way down.

**Comment by rowanhogan** [comment]
So to your first point, it’s one request per page, which is obviously not ideal but it’s our only option at present. I would love to ditch this the second we could get offline amounts in another response. It’s actually more than just offline amounts in that we get a page image and name that the leaderboard response doesn’t include either 😖

So short answer, it’s slow but it’s the best we’ve got right now 😩

To your second point, totally agree, I’ll update. This was clearly a rush job

**Comment by rowanhogan** [comment]
Ok @danhenderson, I've made a few changes here and there... It's not perfect but it ticks a few boxes:

Screen Shot 2020-04-21 at 12 08 01 pm Screen Shot 2020-04-21 at 12 09 52 pm Screen Shot 2020-04-21 at 12 15 40 pm
